test/unit/range/view/view_translation_test.cpp

```diff
@@ -33,100 +33,159 @@
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v3), cmp1);
 
     // reverse translation
-    auto v2 = vec | view::translate_frames(translation_frames::WITH_REVERSE_COMPLEMENT);                         // == [[T,Y,V,R],[Y,V,R,T]]
-    EXPECT_EQ(std::vector<std::vector<aa27> >(v2), cmp2);
+    auto v4 = vec | view::translate(translation_frames::FWD_REV_0);                                            // == [[T,Y,V,R],[Y,V,R,T]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v4), cmp2);
 
     // forward frames translation
-    auto v3 = vec | view::translate_frames(translation_frames::WITH_FRAME_SHIFTS);                       // == [[T,Y,V,R],[R,T,Y,V],[V,R,T]]
-    EXPECT_EQ(std::vector<std::vector<aa27> >(v3), cmp3);
+    auto v5 = vec | view::translate(translation_frames::FWD);                                          // == [[T,Y,V,R],[R,T,Y,V],[V,R,T]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v5), cmp3);
 
     // six frame translation
-    auto v4 = vec | view::translate_frames(translation_frames::SIX_FRAME);   // == [[T,Y,V,R],[R,T,Y,V],[V,R,T],[Y,V,R,T],[T,Y,V,R],[R,T,Y]]
-    EXPECT_EQ(std::vector<std::vector<aa27> >(v4), cmp4);
+    auto v6 = vec | view::translate(translation_frames::SIX_FRAME);        // == [[T,Y,V,R],[R,T,Y,V],[V,R,T],[Y,V,R,T],[T,Y,V,R],[R,T,Y]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v6), cmp4);
+
+    // user-defined frame combination
+    auto v7 = vec | view::translate(translation_frames::FWD_FRAME_0 | translation_frames::FWD_FRAME_2);          // == [[T,Y,V,R],[V,R,T]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v7), cmp5);
 
     // function syntax
-    auto v5 = view::translate_frames(vec, translation_frames::WITH_REVERSE_COMPLEMENT);                          // == [[T,Y,V,R],[Y,V,R,T]]
-    EXPECT_EQ(std::vector<std::vector<aa27> >(v5), cmp2);
+    auto v8 = view::translate(vec, translation_frames::FWD_REV_0);                                             // == [[T,Y,V,R],[Y,V,R,T]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v8), cmp2);
+
+    // combinability
+    auto v9 = vec | view::complement | view::translate(translation_frames::FWD_REV_0);                         // == [[C,M,H,A],[M,H,A,C]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v9), cmp6);
 
     // combinability
-    auto v6 = vec | view::complement | view::translate_frames(translation_frames::WITH_REVERSE_COMPLEMENT);      // == [[C,M,H,A],[M,H,A,C]]
-    EXPECT_EQ(std::vector<std::vector<aa27> >(v6), cmp5);
+    auto v10 = vec | view::complement | view::translate(translation_frames::FWD_REV_0) | ranges::view::take(1);          // == [[C,M,H,A]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v10), cmp7);
+
+    // combinability and function syntax
+    auto v11 = detail::view_translate(view::complement(vec), translation_frames::FWD_REV_0);                   // == [[T,Y,V,R],[Y,V,R,T]]
+    EXPECT_EQ(std::vector<std::vector<aa27> >(v11), cmp6);
 }
```

Comment:
the comments on the right got completely misaligned :laughing: 

Could you make an extra test where you check the conversion to container? And in addition to vector<vector<>> also test concatenated_sequences for the view that returns multiple frames?

In the regular tests you don't need to call `EXPECT_EQ(std::vector<std::vector<aa27> >(v11), cmp6);`, but could instead just call `EXPECT_TRUE((ranges::equal(v11, cmp6)));` (this performs element-wise comparison).
In fact this is preferred so that if someone breaks the conversion to container, only the conversion-to-container-test fails and not every test!